lib/datadog/tracing/configuration/settings.rb

```diff
@@ -165,6 +166,22 @@ def self.extended(base)
                 o.default { env_to_bool(Tracing::Configuration::Ext::ENV_ENABLED, true) }
               end
 
+              # Comma-separated, case-insensitive list of header names that are reported in incoming and outgoing HTTP requests.
+              #
+              # Each header in the list can either be:
+              # * A header name, which is mapped to the respective tags `http.request.headers.<header name>` and `http.response.headers.<header name>`.
+              # * A key value pair, "header name:tag name", which is mapped to the span tag `tag name`.
+              #
+              # You can mix the two types of header declaration in the same list.
+              # Tag names will be normalized based on the [Datadog tag normalization rules](https://docs.datadoghq.com/getting_started/tagging/#defining-tags).
+              #
+              # @default `DD_TRACE_HEADER_TAGS` environment variable, otherwise an empty set of tags
+              # @return [Array<String>]
+              option :header_tags do |o|
+                o.default { env_to_list(Configuration::Ext::ENV_HEADER_TAGS, nil, comma_separated_only: true) }
+                o.setter { |header_tags, _| Configuration::HTTP::HeaderTags.new(header_tags) }
+              end
```

Comment:
If you rebase master on it. We can use the latest configuration options
```suggestion
              option :header_tags do |o|
                o.env Configuration::Ext::ENV_HEADER_TAGS
                o.type :array
                o.default []
                o.setter { |header_tags, _| Configuration::HTTP::HeaderTags.new(header_tags) }
              end
```

If you rebase or merge master, you can use the latest configuration options.